test/suites/edgexfoundry/proxy_test.go

```diff
@@ -0,0 +1,125 @@
+
+	t.Cleanup(func() {
+		require.NoError(t, os.RemoveAll(tmpDir))
+	})
+
+	// Create temp dir for private and public keys
+	require.NoError(t, os.Mkdir(tmpDir, 0755))
+
+	// Get Kong admin JWT token
+	utils.Exec(t, fmt.Sprintf("sudo install -m 604 /var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt ./%s", tmpDir))
+	kongAdminJWTFile := tmpDir + "/kong-admin-jwt"
+	out, err := os.ReadFile(kongAdminJWTFile)
+	require.NoError(t, err)
+	kongAdminJWT := string(out)
+
+	// Generate private and public keys
+	utils.Exec(t, fmt.Sprintf("openssl ecparam -genkey -name prime256v1 -noout -out %s", privateKey))
+	utils.Exec(t, fmt.Sprintf("openssl ec -in %s -pubout -out %s", privateKey, publicKey))
+
+	// Use secrets-config to add a user example with id 1000
+	stdout, _ := utils.Exec(t,
+		fmt.Sprintf("edgexfoundry.secrets-config proxy adduser --token-type jwt --user example --algorithm ES256 --public_key %s --id 1000 -jwt %s",
+			publicKey,
+			kongAdminJWT))
+	// On success, the above command prints the user id
+	require.Equal(t, "1000\n", stdout)
+
+	// Generate a JWT token for the above example user
+	jwt, _ := utils.Exec(t,
+		fmt.Sprintf("edgexfoundry.secrets-config proxy jwt --algorithm ES256 --private_key %s --id 1000 --expiration=1h", privateKey))
+	jwt = strings.TrimSuffix(jwt, "\n")
+
+	// Call an API on behalf of example user
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	req, err := http.NewRequest("GET", "https://localhost:8443/core-data/api/v2/ping", nil)
+	require.NoError(t, err)
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", jwt))
+
+	resp, err := client.Do(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	require.Equal(t, 200, resp.StatusCode)
+}
+
+func TestTLSCert(t *testing.T) {
+
+	const tmpDir = "certs"
+
+	// start clean
+	require.NoError(t, os.RemoveAll(tmpDir))
+
+	t.Cleanup(func() {
+		require.NoError(t, os.RemoveAll(tmpDir))
+	})
+
+	// Create temp dir for certificates and keys
+	require.NoError(t, os.Mkdir(tmpDir, 0755))
+
+	// Get Kong admin JWT token
+	utils.Exec(t, fmt.Sprintf("sudo install -m 604 /var/snap/edgexfoundry/current/secrets/security-proxy-setup/kong-admin-jwt ./%s", tmpDir))
+	kongAdminJWTFile := tmpDir + "/kong-admin-jwt"
+	out, err := os.ReadFile(kongAdminJWTFile)
+	require.NoError(t, err)
+	kongAdminJWT := string(out)
+
+	// Add the certificate, using Kong Admin JWT to authenticate
+	caCertFile, caKeyFile := generateCerts(tmpDir)
+	utils.Exec(t, fmt.Sprintf("edgexfoundry.secrets-config proxy tls --incert %s --inkey %s --admin_api_jwt %s", caCertFile, caKeyFile, kongAdminJWT))
+
+	// Check if TLS is setup correctly returning status code 401
+	code, _ := utils.Exec(t, fmt.Sprintf(`curl --show-error --silent --include --output /dev/null --write-out "%%{http_code}" --cacert %s -X GET 'https://localhost:8443/core-data/api/v2/ping' -H "Authorization: Bearer %s"`, caCertFile, "testToken"))
```

Comment:
We need to make sure the certificate is fully installed before calling the http endpoint. This is how we did it in bash: https://github.com/canonical/edgex-checkbox-provider/blob/ffa6e4477666f5ced2778e106563a2ca4f47eb6d/data/latest/test-security-services-proxy-certs.sh#L146-L156